### PR TITLE
Clean up defunct dependencies

### PR DESCRIPTION
**Pull request by ConnectedSystems**
Used [Aqua](https://github.com/JuliaTesting/Aqua.jl) to help identify issues.

- Added Aqua to tests
- Removed undefined exports
- Moved unused dependencies to [weakdeps] section as they will be used by extension packages later, or
- Removed dependencies altogether as they are no longer needed
- Also bumped the version number in prep for next release

Cleaning this up results in a substantial improvement to precompilation times. Previously it was ~30 seconds for ADRIA.
It is now between 8 - 10 seconds.

```bash
(sandbox) pkg> activate --temp
  Activating new project at `C:\Users\tiwanaga\AppData\Local\Temp\jl_HZP6j3`

# Add ADRIA as a development dependency
(jl_HZP6j3) pkg> dev .
   Resolving package versions...
    Updating `C:\Users\tiwanaga\AppData\Local\Temp\jl_HZP6j3\Project.toml` 

(jl_HZP6j3) pkg> ^C

julia> @time using ADRIA
  8.696201 seconds (9.55 M allocations: 571.700 MiB, 2.02% gc time, 8.13% compilation time: 32% of which was recompilation)
```

Things will only improve as we split functionality out into extension packages.

### CODE REVIEW

**Comment by ConnectedSystems** [comment]
Also note that `OrdinaryDiffEq` is also slated for removal, pending the new model being finalized.

**Comment by Zapiano** [comment]
@ConnectedSystems I tried to run some scenarios and got an error in the line https://github.com/open-AIMS/ADRIA.jl/blob/d1c066453b13e62935d6f13a7bfed4fccf7a65d4/src/scenario.jl#L441

image

**Comment by ConnectedSystems** [comment]
Aaaah my bad, I moved one dependency out after my test run thinking it wasn't needed either. Will fix soon

**Comment by ConnectedSystems** [comment]
Should be all good now!

**Comment by Zapiano** [comment]
Hum, now I got an error when loading the extensions. 

image

**Comment by ConnectedSystems** [comment]
> Hum, now I got an error when loading the extensions.
> 
> image

Could I confirm you updated the ADRIA environment?

**Comment by Zapiano** [comment]
> Could I confirm you updated the ADRIA environment?

Yes, I did... but I didn't restart the REPL after updating. I can try that, give me a second.

**Comment by Zapiano** [comment]
> > Could I confirm you updated the ADRIA environment?
> 
> Yes, I did... but I didn't restart the REPL after updating. I can try that, give me a second.

image

but then, same error when I run `using ADRIA`

image

**Comment by ConnectedSystems** [comment]
Could you do a `status` in the package manager for the ADRIA environment?

**Comment by Zapiano** [comment]
> Could you do a `status` in the package manager for the ADRIA environment?

```
(ADRIA) pkg> status
Project ADRIA v0.12.0
Status `C:\Users\pribeiro\AIMS\Code\ADRIA.jl\Project.toml`
  [c9ce4bd3] ArchGDAL v0.10.4
  [e28b5b4c] Bootstrap v2.4.0
  [336ed68f] CSV v0.10.14
  [aaaa29a8] Clustering v0.15.7
  [861a8166] Combinatorics v1.0.2
  [adafc99b] CpuId v0.3.1
  [a93c6f00] DataFrames v1.6.1
  [864edb3b] DataStructures v0.18.20
  [0703355e] DimensionalData v0.27.3
  [b4f34e82] Distances v0.10.11
  [31c24e10] Distributions v0.25.109
  [cc61a311] FLoops v0.2.1
  [5789e2e9] FileIO v1.16.3
  [62cb38b5] GeoDataFrames v0.3.8
  [68eda718] GeoFormatTypes v0.4.2
  [cf35fbd7] GeoInterface v1.3.4
  [86223c79] Graphs v1.11.1
  [09f84164] HypothesisTests v0.11.0
  [82e4d734] ImageIO v0.6.8
  [a98d9a8b] Interpolations v0.15.1
  [358108f5] JMcDM v0.7.16
  [682c06a0] JSON v0.21.4
  [5cadff95] JuliennedArrays v0.4.0
  [23992714] MAT v0.10.7
  [add582a8] MLJ v0.20.6
  [4744a3fa] ModelParameters v0.4.2
  [30363a11] NetCDF v0.12.0
  [a15396b6] OnlineStats v1.7.0
  [bac558e1] OrderedCollections v1.6.3
  [1dea7af3] OrdinaryDiffEq v6.82.0
  [eebad327] PkgVersion v0.3.3
  [aea7be01] PrecompileTools v1.2.1
  [92933f4c] ProgressMeter v1.10.0
  [189a3867] Reexport v1.2.2
  [05181044] RelocatableFolders v1.0.1
  [cdeec39e] SIRUS v2.0.1
  [0bca4576] SciMLBase v2.41.2
  [efcf1570] Setfield v1.1.1
  [47aef6b3] SimpleWeightedGraphs v1.4.0
  [276daf66] SpecialFunctions v2.4.0
  [860ef19b] StableRNGs v1.0.2
  [90137ffa] StaticArrays v1.9.5
  [2913bbd2] StatsBase v0.34.3
  [4c63d2b9] StatsFuns v1.3.1
  [6fc51010] Surrogates v6.9.0
  [c21b50f5] YAXArrays v0.5.7
  [0a941bbe] Zarr v0.9.4
  [ade2ca70] Dates
  [8ba89e20] Distributed
  [b77e0a4c] InteractiveUtils
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [44cfe95a] Pkg v1.10.0
  [de0858da] Printf
  [9a3f8284] Random
  [2f01184e] SparseArrays v1.10.0
  [10745b16] Statistics v1.10.0
  [fa267f1f] TOML v1.0.3
```

**Comment by ConnectedSystems** [comment]
Oh sorry, I'll also need the versions of GeoMakie, GraphMakie and WGLMakie (or GLMakie, whichever you're using)

**Comment by ConnectedSystems** [comment]
Reverted dependency of GeoMakie to v0.6.x - ADRIA now correctly precompiles for me now.

File: src/scenario.jl [hunk 1]
```diff
@@ -90,7 +90,7 @@ function run_scenarios(
 
     # Cross product between rcps and param_df to have every row of param_df for each rcp
     rcps_df = DataFrame(; RCP=parse.(Int64, RCP))
-    scenarios_df = crossjoin(scens, rcps_df)
+    scenarios_df = crossjoin(scens[:, Not(:RCP)], rcps_df)
```
Comment (Zapiano) [inline src/scenario.jl:93]: Apparently this is raising an error when I run the model

image

**Review by Zapiano — CHANGES_REQUESTED** [review]

File: src/scenario.jl (hunk 1 above)
Comment (ConnectedSystems) [reply src/scenario.jl:93]: :facepalm: let me check what's happened there...

Comment (ConnectedSystems) [reply src/scenario.jl:93]: Okay, I don't know what's happening as it works for me?

But I know which commit is causing it.

Comment (ConnectedSystems) [reply src/scenario.jl]: Could you pull and try again? Issues should be resolved.

Comment (Zapiano) [reply src/scenario.jl]: hum.. now I can `run_model` but I got an error when running the tests (also, it's taking too long to run the tests, I think it was more than 10 mins...that's insane)

image

Comment (ConnectedSystems) [reply src/scenario.jl]: Ah yes Aqua tests will fail as I had to comment out the compat entries.

Strange that it takes 10mins though.

Comment (Zapiano) [reply src/scenario.jl]: Just to clarify, the idea is to leave is this way for now or you are going to fix that in this PR?

Comment (ConnectedSystems) [reply src/scenario.jl]: So I ran it, the tests failed (but we can comment the compat tests out) but it took way less than 10mins.

Comment (Zapiano) [reply src/scenario.jl]: > So I ran it, the tests failed (but we can comment the compat tests out) but it took way less than 10mins.

Hum... I don't know what happened here... 🤔 

I think it's a good idea to comment the compat tests out and open an issue to fix them later. :)

Comment (ConnectedSystems) [reply src/scenario.jl]: I'll do that in an hour or so

Comment (ConnectedSystems) [reply src/scenario.jl]: All done. Tests complete in well under <4mins for me (didn't do a comprehensive timing but based on output)

```julia-repl
Test Summary: | Pass  Total   Time
Aqua          |    4      4  18.8s
Running 64 scenarios for RCP 45 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| Time: 0:01:03        
Test Summary:       | Pass  Total  Time
Temporal clustering |   35     35  0.0s
Test Summary:  | Pass  Total  Time
Domain loading |    4      4  3.8s
Test Summary:        | Pass  Total  Time
Connectivity loading |    3      3  0.3s
Test Summary:      | Pass  Total  Time
Environmental data |    2      2  0.2s
Test Summary:  | Pass  Total  Time
Initial covers |    1      1  0.0s
Test Summary:          | Pass  Total  Time
Cyclone mortality data |    2      2  0.2s
Test Summary:  | Pass  Total   Time
Domain loading |    7      7  32.2s
Test Summary:         | Pass  Total  Time
Truncated normal mean |    2      2  0.0s
Test Summary:        | Pass  Total  Time
Truncated normal cdf |    2      2  0.0s
Test Summary:           | Pass  Total  Time
proportional adjustment |   10     10  1.2s
Test Summary: | Pass  Total  Time
Coral Spec    |   27     27  0.3s
Test Summary: | Pass  Total  Time
Fecundity     |    2      2  0.3s
Test Summary:     | Pass  Total  Time
Larval Production |    1      1  0.8s
Test Summary: | Pass  Total  Time
Recruitment   |    7      7  0.0s
Test Summary: | Pass  Total  Time
growth model  |    6      6  0.4s
Test Summary: | Pass  Total  Time
NetCDF        |    5      5  0.5s
Test Summary: | Pass  Total  Time
axes_names    |    2      2  0.2s
Test Summary: | Pass  Total  Time
sort_axis     |    2      2  0.7s
Test Summary: | Pass  Total  Time
ZeroDataCube  |   12     12  0.6s
Test Summary: | Pass  Total  Time
DataCube      |   21     21  0.9s
Test Summary:           | Pass  Total  Time
Relative Shelter Volume |    6      6  5.8s
Test Summary: | Pass  Total  Time
sample        |   33     33  2.1s
Test Summary:     | Pass  Total  Time
Targeted sampling |   11     11  5.3s
Test Summary: | Pass  Total     Time
Sample bounds |  565    565  1m46.5s
Test Summary: | Pass  Total  Time
Seeding       |   75     75  3.0s
Test Summary:                   | Pass  Total  Time
Extracting component parameters |    4      4  0.5s
Test Summary:                  | Pass  Total  Time
Validate included MCDA methods |    1      1  5.0s
Test Summary: | Pass  Total  Time
text_display  |   28     28  0.0s
Test Summary:    | Pass  Total  Time
Full example run |    2      2  7.8s
     Testing ADRIA tests passed
```

Gotta do something about these excessive warnings that spam the terminal though:

```
┌ Warning: Can not determine size of element type. Using DiskArrays.fallback_element_size[] = 100 bytes
└ @ DiskArrays C:\Users\tiwanaga\.julia\packages\DiskArrays\MpOpv\src\chunks.jl:321
```

**Review by Zapiano — APPROVED** [review]
🚀

**Pull request merged**